Replacing power cells with RPED takes their charge into account

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Instead of get_part_ratings() (which is what the RPED uses to decide whether one stock part is better than another) returning rating * maxcharge on cells, it now returns maxcharge + charge. This means the cell of the highest capacity with the most charge will go in first.

Removes manually set ratings on cells in favour of calculating them in Initialize() in a more consistent and logical way. I don't think it's used for anything now anyway.

![](https://i.imgur.com/2hGw7lW.png)

Other small changes described in the commit messages below.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Machines like the chem dispenser won't work without charged cells in them. This makes it less likely that people will unintentionally swap out charged cells with uncharged ones.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
qol: RPED now swaps in cell with greatest charge (of highest tier) first
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Cells of same tier are replaced with higher charged cells first

carshalash commented:
As a bartender main, this makes my pp hard.

MrMelbert commented on code/modules/power/cell.dm:
```diff
@@ -201,7 +201,8 @@
 
 /obj/item/stock_parts/cell/get_part_rating()
-	return rating * maxcharge
+	rating = max(round(maxcharge/8000,1),1)
```
I feel like this operating shouldn't be done on the `rating` var? You're mutating the actual rating of the cell whenever you call this.

MrMelbert replied:
```suggestion
	rating = max(round(maxcharge / 8000, 1), 1)
```

cacogen committed:
I choose not to name this commit

cacogen committed:
Doesn't override Initialize() to set empty subtypes' charges to 0

- Sets them in the parent cell's Initialize() based on a var instead
- Sets cell rating in Initialize() because it makes more sense than setting them manually (although I left the manually set vars for easy reference)
- Only sets charge to maxcharge if the cell doesn't start with a charge
- Makes the you should not swallow it joke only appear on 10% of cells, to preserve its humour
- Sets the infinite-capacity power cell's maxcharge to INFINITY because examining it and seeing it has a rating of 30 MJ was always really disappointing
- Also sets its rating and chargerate to INFINITY to match
- Removes exclamation mark from infinite-capacity power cell's name because it looks bad at the end of sentences to double punctuation and it should just be impressive rather than insist upon itself
- Makes infinite-capacity power cell's use() return TRUE like it should
- Removes dumb override of Initialize() on crystal_cell that manually sets its charge to be the same as its maxcharge
- Removes manually setting the charge of inducer_supply cell to the same as its maxcharge in favour of letting the parent Initialize() handle it

cacogen committed:
Get rid of manually set ratings values

- Calculate rating in Initialize() in case we need it for something (but do it in logical increments of 10000 so each tier corresponds to another 10000 maxcharge)
- Don't factor rating into getting part ratings because it's just a pointless multiplier when we can just add our maxcharge to our charge

cacogen committed:
Don't set infinite capacity power cell rating manually

- Remove its rating from its desc due to its use of scientific notation when it should just use plain numbers (e.g. 10000000000000000000000000000000000 GJ)
- Calculate part ratings by properly weighting maxcharge so it's not overpowered by charge in cases of lower tier charged cells vs. next tier uncharged cells
